KubeArmor/core/dockerHandler.go

```diff
@@ -176,8 +176,20 @@ func (dm *KubeArmorDaemon) GetAlreadyDeployedDockerContainers() {
 				dm.ContainersLock.Lock()
 				if _, ok := dm.Containers[container.ContainerID]; !ok {
 					dm.Containers[container.ContainerID] = container
-				} else if dm.Containers[container.ContainerID].AppArmorProfile == "" && container.AppArmorProfile != "" {
+				} else if dm.Containers[container.ContainerID].PidNS == 0 && dm.Containers[container.ContainerID].MntNS == 0 {
```

Comment:
```
dm.ContainersLock.Lock()
if _, ok := dm.Containers[container.ContainerID]; !ok {
	dm.Containers[container.ContainerID] = container
} else if dm.Containers[container.ContainerID].PidNS == 0 && dm.Containers[container.ContainerID].MntNS == 0 {
	// this entry was updated by kubernetes before docker detects it
	// thus, we here use the info given by kubernetes instead of the info given by docker

	container.NamespaceName = dm.Containers[container.ContainerID].NamespaceName
	container.ContainerGroupName = dm.Containers[container.ContainerID].ContainerGroupName
	container.ContainerName = dm.Containers[container.ContainerID].ContainerName

	container.PolicyEnabled = dm.Containers[container.ContainerID].PolicyEnabled
	container.ProcessVisibilityEnabled = dm.Containers[container.ContainerID].ProcessVisibilityEnabled
	container.FileVisibilityEnabled = dm.Containers[container.ContainerID].FileVisibilityEnabled
	container.NetworkVisibilityEnabled = dm.Containers[container.ContainerID].NetworkVisibilityEnabled
	container.CapabilitiesVisibilityEnabled = dm.Containers[container.ContainerID].CapabilitiesVisibilityEnabled

	dm.Containers[container.ContainerID] = container

	for _, conGroup := range dm.ContainerGroups {
		if conGroup.ContainerGroupName == container.ContainerGroupName && conGroup.AppArmorProfiles[container.ContainerID] == "" {
			conGroup.AppArmorProfiles[container.ContainerID] = container.AppArmorProfile
			break
		}
	}
} else {
	dm.ContainersLock.Unlock()
	continue
}
dm.ContainersLock.Unlock()
```